### ISSUE TITLE
Getting CeleryNotRunningException repeatedly

### ISSUE THREAD

**Issue by jakeczyz** [onnela-lab/beiwe-backend#158]
As of a few days ago, my cluster is reporting CeleryNotRunningException (raised from line 242 in services/celery_data_processing.py) repeatedly via Sentry (240 times per day).  From my perusal of the code comments in that module, it's not clear if this is a problem or an acceptable state of the system.  Any advice would be appreciated!

**Comment by biblicabeebli** [comment]
Celery is a task dispatcher library/platform that we use for managing data processing tasks.
Due to problems with RabbitMQ the CeleryNotRunningException can occur randomly when the server is under a heavy load.  There is an issue to replace RabbitMQ with another broker (#93).

In this case you seem to have had a total celery failure.  This is because you have a consistent number of errors per hour.

Easiest solution here is for you to terminate your data processing server and deploy a new one.

(Usually this is caused by out of disk space errors, there are impovements queued onthe

**Comment by jakeczyz** [comment]
Thanks @biblicabeebli. Is it sufficient to reboot the data processing manager EC2 instance, or is there a better approach to `terminate your data processing server and deploy a new one.`?

**Comment by biblicabeebli** [comment]
Its not possible for me to answer that without knowing what occurred on the server; I am only able to make well informed guesses.

minor clarification:

> This is because you have a consistent number of errors per hour

By this I mean that constant number of these errors per hour means a total celery failure.  If there is variation it means that the system is heavily loaded but doing okay.

**Comment by biblicabeebli** [comment]
And also....

The size of the server storage has been increased on the terminally delayed development branch.

Closing issue because no further action can be taken within this forum.

**Comment by jakeczyz** [comment]
Thanks for your responses and tips.  My update, in case it helps anyone in similar shoes: I rebooted the dp manager instance, via the EC2 console (I don't run a worker as a cost-saving measure, but, presumably, running one would reduce load on the manager...).  I also bumped the EB filesystem setting to use a 15 GB disk instead of the default (which is 8) (see tip here: https://github.com/onnela-lab/beiwe-backend/issues/143#issuecomment-657652715). After last night's nightly EB instance replacement, the errors seem to have stopped.
